File: Frontend/secure-booking-web/src/types/Hotel.ts
export enum PropertyType {
  Hotel = "Hotel",
  Resort = "Resort",
  Villa = "Villa",
  Apartment = "Apartment",
}

export interface Hotel {
  id: string;
  name: string;
  description?: string;
  starRating: number;
  reviewScore: number;
  propertyType: PropertyType;
  amenities: string[];
  imageUrl?: string;
  isActive: boolean;
  locationId: string;
  locationCity: string;
  locationCountry: string;
  roomCount: number;
  minPricePerNight?: number;
  createdAt: string;
}

File: Backend/SecureBooking.Domain/Entities/Hotel.cs
namespace SecureBooking.Domain.Entities;

using SecureBooking.Shared.Enums;

public class Hotel : Entity
{
    public string Name { get; set; } = string.Empty;
    public string? Description { get; set; }

    public int StarRating { get; set; } // 1-5

    public double ReviewScore { get; set; } // 0-10 guest review score

    public PropertyType PropertyType { get; set; } = PropertyType.Hotel;

    public ICollection<string> Amenities { get; set; } = [];

    public string? ImageUrl { get; set; }

    public bool IsActive { get; set; } = true;

    public Guid LocationId { get; set; }
    public Location? Location { get; set; }

    public ICollection<Room> Rooms { get; set; } = [];
}